pkg/generate/git/repository.go

```diff
@@ -0,0 +1,124 @@
+package git
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"os/exec"
+	"regexp"
+	"strings"
+	"unicode"
+)
+
+// execCmdFunc is a function that executes an external command
+type execCmdFunc func(dir, name string, args ...string) (string, string, error)
```

Comment:
It's for testing. You can pass in a fake exec function to test the logic in repository.